neo/rawio/spikegadgetsrawio.py

```diff
@@ -103,7 +123,21 @@ def _parse_header(self):
 
         self._sampling_rate = float(hconf.attrib["samplingRate"])
         num_ephy_channels = int(hconf.attrib["numChannels"])
-
+        
+        # check for agreement with number of channels in xml
+        sconf_channels = np.sum([len(x) for x in sconf])
+        if sconf_channels < num_ephy_channels:
+            num_ephy_channels = sconf_channels
+        if sconf_channels > num_ephy_channels:
+            raise ValueError(
```

Comment:
```suggestion
            raise NeoReadWriteError(
```

I think a value error would imply that the user could fix something. Otherwise a NeoReadWriteError is the way to tell the end-user that this is a file that does not fit with the Neo model. We may need to add this to the imports.

Reply:
Yes I just check we need to import the error message!

with 
```python
from neo.core import NeoReadWriteError
```

at the top.